### PR TITLE
Aragon voting page

### PR DESCRIPTION
**Pull request by kozlovzxc-at-lido**
## Description

Added ability to vote on Aragon with locked tokens.

## Demo

Incorrect form
image

Working form 
image

Notification if voting is closed
image

Notification if voting doesn't exist
Screenshot 2023-04-25 at 12 36 25

## Review notes

Check testing steps if you want to run locally

## Testing notes

Testing steps https://www.notion.so/TRP-UI-test-cases-5251b40bd1b64fc183165c2e03d7f7d1?pvs=4

### CODE REVIEW

File: features/aragon/aragonForm.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import { useGetVoting } from './useAragon';
+
+type AragonFormData = {
+  voteId: string;
+  success: boolean;
+};
+
+const validateVoteId = (value: string) => {
+  const number = Number(value);
+  if (number.toString() !== value) {
```
Comment (Jabher) [inline features/aragon/aragonForm.tsx:19]:
```suggestion
  if (!Number.isNaN(number)) {
```

File: features/aragon/aragonForm.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import { stringToNumber } from 'shared/lib';
+import { useGetVoting } from './useAragon';
+
+type AragonFormData = {
+  voteId: string;
+  success: boolean;
+};
+
+const validateVoteId = (value: string) => {
+  const number = Number(value);
```
Comment (Jabher) [inline features/aragon/aragonForm.tsx:18]:
```suggestion
  const number = parseInt(value);
```

as you do not want to accidentally get floats

File: features/aragon/aragonForm.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      if (vote == null) {
+        ToastError(`Voting doesn't exists`);
+        return;
+      }
+      if (vote?.open === false) {
+        ToastError('Voting is closed');
+        return;
+      }
+
+      const callData = await encodeCalldata(stringToNumber(voteId), success);
```
Comment (Jabher) [inline features/aragon/aragonForm.tsx:54]: why you even need stringToNumber if you have parseInt/parseFloat?

File: features/aragon/useAragon.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  [CHAINS.Mainnet]: '0x2e59A20f205bB85a89C53f1936454680651E618e',
+  [CHAINS.Goerli]: '0xbc0B67b4553f4CF52a913DE9A6eD0057E2E758Db',
+};
+
+export const useAragon = () => {
+  const { chainId } = useWeb3();
+
+  // TODO: get this from env
+  const aragonAddress =
+    chainId != null ? aragonAddressMap[chainId as CHAINS] : '';
```
Comment (Jabher) [inline features/aragon/useAragon.ts:18]: looks like you're not using our prettier config, 120 chars should fit this in single line

File: pages/aragon.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      <AragonForm />
+    </Layout>
+  );
+};
+
+export default Aragon;
+
+export const getStaticProps: GetStaticProps = async () => ({
+  props: {},
+});
```
Comment (Jabher) [inline pages/aragon.tsx:23]: ???

**Review by Jabher — APPROVED** [review]

File: features/aragon/aragonForm.tsx (hunk 2 above)
Comment (kozlovzxc-at-lido) [reply features/aragon/aragonForm.tsx:18]: Thanks, going to fix 👍

File: features/aragon/aragonForm.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    formState: { isValid, errors },
+  } = useForm<AragonFormData>({ mode: 'onChange' });
+
+  const encodeCalldata = useEncodeAragonCalldata();
+  const aragonVote = useAragonVote();
+  const getVoting = useGetVoting();
+
+  const runTransaction = useCallback(
+    async (data: AragonFormData) => {
+      const { voteId, success } = data;
```
[the remark this answers is not in the document]
Comment (kozlovzxc-at-lido) [reply features/aragon/aragonForm.tsx:42]: Yeah, right 👍

File: features/aragon/aragonForm.tsx (hunk 3 above)
Comment (kozlovzxc-at-lido) [reply features/aragon/aragonForm.tsx:54]: yeah, I don't need it, going to simplify

File: features/aragon/aragonForm.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  return (
+    <Block>
+      <Form onSubmit={handleSubmit(runTransaction)}>
+        <InputGroupStyled fullwidth error={errors.voteId?.message?.toString()}>
+          <SelectVesting error={errors.voteId != null} />
+          <InputNumber
+            fullwidth
+            label="Vote ID"
+            error={errors.voteId != null}
+            {...register('voteId', {
```
[the remark this answers is not in the document]
Comment (kozlovzxc-at-lido) [reply features/aragon/aragonForm.tsx:77]: It's official recommendation how to use it https://react-hook-form.com/get-started/

File: features/aragon/useAragon.ts (hunk 4 above)
Comment (kozlovzxc-at-lido) [reply features/aragon/useAragon.ts:18]: Hm, it's actually not using shared config, will add one in other PR

File: pages/aragon.tsx (hunk 5 above)
Comment (kozlovzxc-at-lido) [reply pages/aragon.tsx]: Discussed in DM, going to remove async

**Review by Jabher — APPROVED** [review]

**Pull request merged**